cpp/lib/src/SmartDownloader.cc

```diff
@@ -189,56 +188,22 @@ bool DiglitSmartDownloader::downloadDocImpl(const std::string &url, const TimeLi
                                             std::string * const document, std::string * const http_header_charset,
                                             std::string * const error_message)
 {
+    std::string url_improved(url);
+    if (RegexMatcher::Matched("/diglit/", url))
+        url_improved = StringUtil::ReplaceString("/diglit/", "/opendigi/", &url_improved);
+    if (RegexMatcher::Matched("/opendigi/", url_improved))
+        url_improved = url_improved + "/ocr";
```

Comment:
Please use +=, it's more efficient and also easier for humans to parse.

Reply:
done